### DISCUSSION TITLE
Building a UI for fedimint module, where to start?

### DISCUSSION THREAD

**Discussion by toyota-corolla0** [fedimint/fedimint#3552] [Q&A] [answered]
I am excited to say that I am ready to start building a UI for my [prediction markets](https://github.com/toyota-corolla0/fedimint-prediction-markets) module. I am not sure where to start with this and would appreciate some advice. Thanks!!!

**Comment by elsirion** [comment, marked as answer]
You could try to patch it into webimint: https://github.com/elsirion/webimint-rs

There isn't really a standard way of implementing module UIs yet, probably because there are barely any Fedimint UIs (I'm aware of Fedi and webimint, we should curate a list). For ideas how it might work as a Fedi mod @wbobeirne or @justinmoon might be able to chime in.

**Comment by wbobeirne** [comment]
Can you describe what your module does, and what you'd like a UI to do? It's a little hard to make specific recommendations without knowing a bit more. If you want a UI for administrating your module, understanding the [gateway UI](https://github.com/fedimint/ui/tree/master/apps/gateway-ui) and [gateway server API](https://github.com/fedimint/fedimint/blob/master/gateway/ln-gateway/src/rpc/rpc_server.rs) would be a good place to start. If you're trying to make a client-focused module, then webimint might be your best bet.

**Comment by dpc** [comment]
BTW. I'd start with implementing [`ClientModule::handle_cli_command`](https://github.com/fedimint/fedimint/blob/6521604f808fe3cf2c405749cb08effc82731560/fedimint-client/src/module/mod.rs#L104). It will allow you to build `fedimint-cli` with support for interacting with the module which is currently the only standard "UI".

**Comment by toyota-corolla0** [comment]
@elsirion Webimint is exactly what I was looking for! thanks!

@wbobeirne I should have specified, the client side UI is what I want to work on. [quick explanation of my module](https://stacker.news/items/294781). 

@dpc I have implemented ```ClientModule::handle_cli_command``` and the module seems to be all working as intended. I have not written any test yet though and will need to start working on that.

Thanks for the replies!

**Reply by EthnTuttle** [reply]
when you've made some more progress, perhaps you can commit you project to this awesome list: https://github.com/EthnTuttle/awesome-ecash/
